src/google/protobuf/compiler/php/php_generator.cc

```diff
@@ -726,6 +726,11 @@ void GenerateFieldAccessor(const FieldDescriptor* field, bool is_descriptor,
       printer->Print(");\n");
     }
   } else if (field->cpp_type() == FieldDescriptor::CPPTYPE_MESSAGE) {
+    if (IsWrapperType(field)) {
```

Comment:
In this way, you also need to implement GPBWrapperTypes in c extension.
I was thinking we could simply do the box/unbox implementation in the setter/getter.
```
public function setStringValue($val) {
  if (is_string($va)) {
    $val = new Google\Protobuf\StringValue(["value"=>$val]);
  }
  GPBUtil::checkMessage($val, Google\Protobuf\StringValue::class)
  $this->string_value_field = $val;
}
```

Reply:
Since we have to support the same logic for repeated field, I would suggest changing checkMessage which is also shared by repeated field.

Reply:
Isn't that what I am doing? I am using IsWrapperType check in php_generator.cc, and using it to add the extra `"$var = GPBWrapperTypes::normalizeToMessageType($var, \\^class_name^::class);\n",` line into the generated setter.

Re `checkMessage`, if it was implemented that way, would I also need to update the c extension implementation? (here: https://github.com/protocolbuffers/protobuf/blob/master/php/ext/google/protobuf/type_check.c#L413). That was why I didn't take the approach of modifying `checkMessage`, and instead added code to the setters.

Reply:
I can implement the c extension part. Thanks for writing the test part already.

Reply:
No worries! I have updated the PR to add the check in to checkMessage, and added a single test case for repeated fields. If this looks good, I will go ahead and update generated phpdocs and add complete test coverage.